samza-sql/src/main/java/org/apache/samza/sql/data/SamzaSqlRelMessage.java

```diff
@@ -0,0 +1,74 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.apache.samza.sql.data;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+
+/**
+ * Samza sql relational message. Each Samza sql relational message represents a relational row in a table.
+ * Each row of the relational table and hence SamzaSqlRelMessage consists of list of column values and
+ * their associated column names. Right now we donot store any other metadata other than the column name in the
+ * SamzaSqlRelationalMessage, In future if we find a need, we could add additional column ddl metadata around
+ * primary Key, nullability, etc.
+ */
+public class SamzaSqlRelMessage {
+
+  private List<Object> value;
+  private List<String> names;
+
+  public SamzaSqlRelMessage(List<Object> values, List<String> names) {
```

Comment:
Makes sense, Key is optional since not all the SamzaSQlRelMessages will have the key field.
I will make it like this
SamzaSqlRelMessage(Optional<Object> key, List values, List names).